libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1116,13 +1116,9 @@ class MapboxNavigation @VisibleForTesting internal constructor(
      * By default uses [MapboxRerouteController].
      */
     fun setRerouteController(rerouteController: RerouteController) {
-        val currentController = this.rerouteController
-        this.rerouteController = LegacyRerouteControllerAdapter(rerouteController)
-
-        if (currentController?.state == RerouteState.FetchingRoute) {
-            currentController.interrupt()
-            reroute()
-        }
+        setRerouteController(
+            LegacyRerouteControllerAdapter(rerouteController) as NavigationRerouteController?
```

Comment:
I described this problem in Slack, TLDR: without this cast kotlin will think that the call is ambiguous and won't be able to choose between `fun setRerouteController(rerouteController: RerouteController)` and `fun setRerouteController(rerouteController: NavigationRerouteController? = defaultRerouteController)`. I think that's an API design bug.

Reply:
```suggestion
            LegacyRerouteControllerAdapter(rerouteController) as NavigationRerouteController
```

Reply:
No, that's the point! It has to be nullable in order to avoid ambiguity. That's why I don't like it so much.

Reply:
nice catch actually! thanks to add `set reroute controller`